Update test.yml on new branch

kassabry opened the pull request:
The old branch will be removed. This should maintain the history of the test.yml file that was lost with the other branch creation.

kassabry committed:
Update test.yml

agitter commented:
Do we know how to work around this error:
```
UnboundLocalError: local variable 'chunk' referenced before assignment
```
I believe we discussed that on one of our recent calls.

The Windows and macOS jobs are cancelled when the Ubuntu job hits an error. I believe we can try setting fail-fast to false in the matrix as described [here](https://stackoverflow.com/questions/70830094/marking-github-actions-workflow-as-failed-if-a-single-job-fails-in-matrix/70841204#70841204).

kassabry committed:
Update test.yml

Updated the BT_Clustering code line

kassabry committed:
Update test.yml

kassabry committed:
Update test.yml

Added the unzipping to the github actions which I believe was causing the problem with chunk referenced before assignment

agitter committed:
Disable fail fast

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast

agitter commented:
@kassabry, @Malnammi got the tests passing on the master branch. Should we revisit this test you added in this pull request? We would need to merge in the new test updates from master.
